processor/removeemptyvaluesprocessor/processor.go

```diff
@@ -0,0 +1,283 @@
+	return ok
+}
+
+func (evp *emptyValueProcessor) SkipBody() bool {
+	// If only the field is specified, but no trailing key, the whole resource should be skipped
+	_, ok := evp.excludeBodyKeySet[""]
+	return ok
+}
+
+func (evp *emptyValueProcessor) processTraces(_ context.Context, td ptrace.Traces) (ptrace.Traces, error) {
+	resourceSpans := td.ResourceSpans()
+	for i := 0; i < resourceSpans.Len(); i++ {
+		resourceSpan := resourceSpans.At(i)
+		scopeSpans := resourceSpan.ScopeSpans()
+
+		if !evp.SkipResourceAttributes() {
+			cleanMap(resourceSpan.Resource().Attributes(), evp.c, evp.excludeResourceKeySet)
+		}
+
+		if evp.SkipAttributes() {
+			// Skip loops for attributes if we don't need to clean them.
+			continue
+		}
+
+		for j := 0; j < scopeSpans.Len(); j++ {
+			scopeSpan := scopeSpans.At(j)
+			spans := scopeSpan.Spans()
+
+			for k := 0; k < spans.Len(); k++ {
+				span := spans.At(k)
+				cleanMap(span.Attributes(), evp.c, evp.excludeAttributeKeySet)
+			}
+		}
+	}
+
+	return td, nil
+}
+
+func (evp *emptyValueProcessor) processLogs(_ context.Context, ld plog.Logs) (plog.Logs, error) {
+	resourceLogs := ld.ResourceLogs()
+	for i := 0; i < resourceLogs.Len(); i++ {
+		resourceLog := resourceLogs.At(i)
+		scopeLogs := resourceLog.ScopeLogs()
+
+		if !evp.SkipResourceAttributes() {
+			cleanMap(resourceLog.Resource().Attributes(), evp.c, evp.excludeResourceKeySet)
+		}
+
+		for j := 0; j < scopeLogs.Len(); j++ {
+			scopeLog := scopeLogs.At(j)
+			logRecords := scopeLog.LogRecords()
+
+			for k := 0; k < logRecords.Len(); k++ {
+				logRecord := logRecords.At(k)
+				if !evp.SkipAttributes() {
+					cleanMap(logRecord.Attributes(), evp.c, evp.excludeAttributeKeySet)
+				}
+
+				if !evp.SkipBody() {
+					cleanLogBody(logRecord, evp.c, evp.excludeBodyKeySet)
+				}
+			}
+		}
+	}
+
+	return ld, nil
+}
+
+func (evp *emptyValueProcessor) processMetrics(_ context.Context, md pmetric.Metrics) (pmetric.Metrics, error) {
+	resourceMetrics := md.ResourceMetrics()
+	for i := 0; i < resourceMetrics.Len(); i++ {
+		resourceMetric := resourceMetrics.At(i)
+		scopeMetrics := resourceMetric.ScopeMetrics()
+
+		if !evp.SkipResourceAttributes() {
+			cleanMap(resourceMetric.Resource().Attributes(), evp.c, evp.excludeResourceKeySet)
+		}
+
+		if evp.SkipAttributes() {
+			// Skip loops for attributes if we don't need to clean them.
+			continue
+		}
+
+		for j := 0; j < scopeMetrics.Len(); j++ {
+			scopeMetric := scopeMetrics.At(j)
+			metrics := scopeMetric.Metrics()
+
+			for k := 0; k < metrics.Len(); k++ {
+				metric := metrics.At(k)
+				cleanMetricAttrs(metric, evp.c, evp.excludeAttributeKeySet)
+			}
+		}
+	}
+	return md, nil
+}
+
+// cleanMap removes empty values from the map, as defined by the config.
+func cleanMap(m pcommon.Map, c Config, excludeKeys map[string]struct{}) {
+	m.RemoveIf(func(s string, v pcommon.Value) bool {
+		for key := range excludeKeys {
+			if key == s {
+				return false
+			}
+		}
```

Comment:
Since this is a map now I think you can just do a look up.

```go
if _, ok := excludeKeys[key]; ok {
  return false
}
```

Reply:
good catch, kinda defeats the whole purpose without that, huh?